contracts/core/CrvDepositorWrapperForwarder.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import { IERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts-0.8/token/ERC20/utils/SafeERC20.sol";
+import { IBalancerVault } from "../interfaces/balancer/IBalancerCore.sol";
+import { ICrvDepositorWrapper } from "../interfaces/ICrvDepositorWrapper.sol";
+import { ICrvDepositor } from "../interfaces/ICrvDepositor.sol";
+import { BalInvestor } from "./BalInvestor.sol";
+
+/**
+ * @title   CrvDepositorWrapperForwarder
+ * @notice  Converts BAL -> balBPT, then wraps to auraBAL via the crvDepositor.
+ *          Finally it forwards the minted auraBAL to a given address.
+ */
+contract CrvDepositorWrapperForwarder is ICrvDepositorWrapper, BalInvestor {
+    using SafeERC20 for IERC20;
+
+    address public immutable crvDeposit;
+    address public immutable cvxCrv;
+    address public immutable forwardTo;
+
+    constructor(
+        address _crvDeposit,
+        IBalancerVault _balancerVault,
+        address _bal,
+        address _weth,
+        bytes32 _balETHPoolId,
+        address _cvxCrv,
+        address _forwardTo
+    ) BalInvestor(_balancerVault, _bal, _weth, _balETHPoolId) {
+        crvDeposit = _crvDeposit;
+        forwardTo = _forwardTo;
+        cvxCrv = _cvxCrv;
+    }
+
+    function setApprovals() external {
+        _setApprovals();
+        require(IERC20(BALANCER_POOL_TOKEN).approve(crvDeposit, type(uint256).max), "!approval");
+    }
+
+    /**
+     * @dev Gets minimum output based on BPT oracle price
+     * @param _amount Units of BAL to deposit
+     * @param _outputBps Multiplier where 100% == 10000, 99.5% == 9950 and 98% == 9800
+     * @return minOut Units of BPT to expect as output
+     */
+    function getMinOut(uint256 _amount, uint256 _outputBps) external view returns (uint256) {
+        return _getMinOut(_amount, _outputBps);
+    }
+
+    /**
+     * @dev Mints auraBal and transfers it to `forwardTo` address
+     * @param _amount Units of BAL to deposit
+     * @param _minOut Min amount of auraBal to be deposited.
+     * @param _lock Should be always true
+     * @param _stakeAddress Hast to be zero address, to ensure cvxCrv is minted only.
+     */
+    function deposit(
+        uint256 _amount,
+        uint256 _minOut,
+        bool _lock,
+        address _stakeAddress
+    ) external {
+        require(_stakeAddress == address(0), "!_stakeAddress");
```

Comment:
To be explicit, if the stakeAddress is not zero, then no cvxCrv is returned. 

+  This contract should only be used by the AuraStakingProxy. 

```
    function _distribute(uint256 _minOut) internal {
        //convert crv to cvxCrv
        uint256 crvBal = IERC20(crv).balanceOf(address(this));
        if (crvBal > 0) {
            uint256 minOut = _minOut != 0
                ? _minOut
                : ICrvDepositorWrapper(crvDepositorWrapper).getMinOut(crvBal, outputBps);
            ICrvDepositorWrapper(crvDepositorWrapper).deposit(crvBal, minOut, true, address(0));
        }

```